llvm/lib/Target/AMDGPU/AMDGPUCallLowering.cpp

```diff
@@ -233,6 +241,13 @@ struct AMDGPUOutgoingArgHandler : public AMDGPUOutgoingValueHandler {
   void assignValueToReg(Register ValVReg, Register PhysReg,
                         const CCValAssign &VA) override {
     MIB.addUse(PhysReg, RegState::Implicit);
+
+    if (VA.getLocVT() == MVT::i1 &&
```

Comment:
For the i1 inreg case, do you get getLocVT == MVT::i32? that should still hit the extendRegisterMin32 path